doc/functional-testing.md

```diff
@@ -0,0 +1,230 @@
+
+Any error from `.findLink()` will reject the returned promise.  Otherwise, the promise is resolved or rejected by `.requestAsync()`.
+
+    it("requests the health check service", () => {
+        return test.startAsync().then((response) => {
+            return test.follow(response, "service", "health-check");
+        });
+    });
+
+
+### `* = test.getHeader(headers, headerName)`
+
+Searches the `headers` object for a case-insensitive match to `headerName`.  This is necessary because the headers are technically case insensitive but JavaScript has case sensitive object property names and case sensitive string comparisons.
+
+When there is no match, `null` is returned instead.
+
+    var headers;
+
+    headers = {
+        "Content-Type": "text/plain"
+    };
+    console.log(test.getHeader(headers, "Content-Type")); // works
+    console.log(test.getHeader(headers, "content-type")); // works
+    console.log(test.getHeader(headers, "CONTENT-TYPE")); // works
+    console.log(test.getHeader(headers, "ContentType")); // fail
+
+
+### `Promise<response> = test.loginAsync()`
+
+Initiates a login.  If there is no account created yet, this first creates an account.  Assigns `this.state.sessionId`, which can be used manually to simulate requests that occur during a logged in session.
+
+Checks for errors along the way and rejects the promise when anything goes wrong.
+
+    it("simulates logged in sessions", () => {
+        return test.loginAsync().then((response) => {
+            return this.followAsync(response, "up", "account", {
+                headers: {
+                    Cookie: `login=${test.state.sessionId}`
+                }
+            });
+        });
+    });
+
+
+### `test.reformatRequestOptions(options)`
+
+Modifies the `options` object so it is in an expected format for the mock request.  Sets several properties and defaults many values.  Assists with URI generation by replacing parameters and appending query string values.
+
+Probably not something that should be called externally.
+
+
+### `Promise<response> = test.requestAsync(options)`
+
+Initiates the fake request.  It does this by first setting up a couple mocks and changing the options that were supplied into a set that the mocks require.
+
+From here, it makes a promise, fixes a few problems that are created by a conflict between Restify and the mocks, then starts the request.  Events are simulated to send any body on the request and the response events are captured to resolve the promise.  A timer is also started and that timer will help ensure that tests do not run for too long.  The promise is resolved if there is any response and is rejected only if the timer expires.
```

Comment:
Don't need the first "that are"

The next to the last sentence can be rewritten as, "A timer is also started, helping to ensure tests do not run for too long."